### PR TITLE
refactor: Remove gogoproto yaml annotations (#409)

### PR DESCRIPTION
**Pull request by ruhatch**
## Description

Remove gogoproto yaml annotations.

Closes: #409 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by clevinson** [comment]
Looks good! @ruhatch is there a reason we're adding `json_name` in regen ledger, but not in the SDK? I'm looking at https://github.com/cosmos/cosmos-sdk/pull/9780 as a corollary to this work, but didn't see any `json_name` tag additions in that PR.

File: proto/regen/ecocredit/v1alpha1/types.proto [hunk 1]
```diff
@@ -118,5 +120,5 @@ message CreditTypeSeq {
   string abbreviation = 1;
 
   // The sequence number of classes of the credit type
-  uint64 seq_number = 2;
+  uint64 seq_number = 2 [ json_name = "seq_number" ];
```
[the remark this answers is not in the document]
Comment (clevinson) [reply proto/regen/ecocredit/v1alpha1/types.proto:123]: Historically this was about multiple word fields, and ensuring yaml encoding used snake-case instead of camel-case.

[the remark this answers is not in the document]
Comment (clevinson) [reply proto/regen/ecocredit/v1alpha1/types.proto:123]: I'm actually doing some further digging into edge cases and how this works here, as I'm suspicious there may be some strange change in behavior as previously things might have been json serialized as camel case...

**Comment by clevinson** [comment]
So digging into it a little more... I'm not actually sure if we need `json_name` field explicitly set at all, or what benefit we get from that?

Most of the time when we marshal/unmarshal (at least in the SDK), we're jsonpb, and serializing using the original proto field name (we do this by setting the [Marshaler.OrigName](https://github.com/cosmos/cosmos-sdk/blob/master/codec/json.go#L21) to `true`).  Because of this, all serialization/deserialization using jsonpb should already work as expected and correctly parse/serialize as the field name (not the camelCase version).

One side note- we actually don't use the codec marshaler when creating the template JSON in our `regen gen-batch-json` cli command (I'll file an issue for this shortly).

By explicitly setting `json_name` here, we force any proto deserializers to fail when parsing camelCase fields, which actually goes against the standard behavior of proto as described in the [language guide](https://developers.google.com/protocol-buffers/docs/proto3#json):

> Parsers accept both the lowerCamelCase name (or the one specified by the json_name option) and the original proto field name.

If there's a specific reason for us wanting to fail on parsing camelCase JSON, then I think we should keep the tags. But otherwise I think adding the explicit `json_name` tags is just yet another thing to remember and maybe we should just drop it entirely, relying instead on the fact that we're already setting `Marshaler.OrigName = true` in the jsonpb codec.

**Comment by ruhatch** [comment]
> Looks good! @ruhatch is there a reason we're adding `json_name` in regen ledger, but not in the SDK? I'm looking at [cosmos/cosmos-sdk#9780](https://github.com/cosmos/cosmos-sdk/pull/9780) as a corollary to this work, but didn't see any `json_name` tag additions in that PR.

Okay, interesting! I need to check whether the new yaml library (which is only used in one place), will use the correct kind of JSON serialization or not.

I'll remove those tags!

**Review by likhita-809 — APPROVED** [review]
lgtm

File: x/ecocredit/client/tx.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/cosmos/cosmos-sdk/client/tx"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/spf13/cobra"
-	"gopkg.in/yaml.v2"
+	"sigs.k8s.io/yaml"
```
Comment (cyberbono3) [inline x/ecocredit/client/tx.go:16]: why this change?

**Review by cyberbono3 — APPROVED** [review]
